package.json

```diff
@@ -53,8 +53,18 @@
     "@typescript-eslint/eslint-plugin": "^3.10.1",
     "@typescript-eslint/parser": "^3.10.1",
     "eslint": "^7.8.1",
+    "husky": "^4.3.0",
+    "lint-staged": "^10.3.0",
     "nodemon": "^2.0.4",
     "prettier": "^2.1.1",
     "typescript": "^4.0.2"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.ts": ["npx eslint --fix", "prettier --no-config --write", "npx tsc", "git add dist"]
```

Comment:
oh i meant literally `"*"` (to capture every staged file). is there a reason to filter to just ts files?

Reply:
nitpick: can we do `"git add dist/"` (with the slash)

Reply:
> oh i meant literally `"*"` (to capture every staged file). is there a reason to filter to just ts files?
> 
> my reason for `"*"` is because that's now `npm run format` works (e.g. `npx eslint --fix . && prettier --no-config --write .` the periods `.`)

I can't do that because `lint-staged` gives the `.ts` file to `npx tsc` so if I change it to work with all files, it would fail because `npx tsc` cant compile `.json`.